Feature/grant recipe list

wenqianli150 opened the pull request:
## Description

_Updated features grant recipe list page_


## What type of PR is this? (check all applicable)

- [ ] Feature


##  Screenshots, Recordings
double click description  could link to detail pages
<img width="1426" height="424" alt="Screenshot 2025-11-04 at 1 11 39 PM" src="https://github.com/user-attachments/assets/fb4c3044-e81b-4ba8-9b2f-ba943dc426a6" />
<img width="1074" height="268" alt="image" src="https://github.com/user-attachments/assets/0dd10109-4181-4119-8044-2462720599e4" />

shardareddy12 committed:
feat: Set up routing for grant recipes

- Added basic list and detail pages for grant recipes
- Updated `routes.tsx` with new navigation paths
- Added menu items in `TemplateConfig.tsx`
- Organized page structure for grant management

Commit:
Update Grant Recipes List Page layout and functionality

Commit:
Update Grant Recipes List Page layout and functionality

Commit:
Update Grant Recipes List Page layout and functionality

Commit:
fix bug with firebase

ibarrasb commented:
@wenqianli150 you need to create the pull request into dev, not main

wenqianli150 committed:
Merge branch 'dev' into feature/grant_recipe_list

jnakaso commented on src/services/grantRecipeService.ts:
```diff
@@ -63,6 +66,29 @@ class GrantRecipeService extends FirestoreService<GrantRecipe> {
   }
+
+  // Fetch all recipes
+  async findAll(): Promise<GrantRecipe[]> {
```
@digitalaidseattle/firebase has a getAll method that does the same thing

jnakaso commented on src/services/grantRecipeService.ts:
```diff
@@ -63,6 +66,29 @@ class GrantRecipeService extends FirestoreService<GrantRecipe> {
+  }
+
+  // Delete a recipe
+  async delete(entityId: Identifier): Promise<void> {
```
why reimplement the method?

jnakaso commented on src/services/grantRecipeService.ts:
```diff
@@ -31,14 +32,16 @@ class GrantRecipeService extends FirestoreService<GrantRecipe> {
   async insert(entity: GrantRecipe, select?: string, user?: User): Promise<GrantRecipe> {
     if (!user?.email) throw new Error("grantRecipeService.insert: user.email is required");
     const now = new Date();
+    // Remove id field as Firestore will auto-generate it
```
This is easier to read

```
const decorated: GrantRecipe = {
{
        ...entity,
        createdAt: now,
        updatedAt: now,
        createdBy: user.email,
        updatedBy: user.email,
} 

delete decorated.id;

super.insert(decorated);
```

jnakaso commented on src/services/grantProposalService.ts:
```diff
@@ -55,6 +56,29 @@ class GrantProposalService extends FirestoreService<GrantProposal> {
       user
     );
   }
+
```
@digitalaidseattle/firebase has a getAll method that does the same thing

jnakaso commented on src/pages/grants/GrantRecipesListPage.tsx:
```diff
@@ -1,17 +1,362 @@
   const navigate = useNavigate();
+  const { user } = useContext(UserContext);
+  const [recipes, setRecipes] = useState<GrantRecipe[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
```
The application has a LoadingContextProvider.  Please us it.

jnakaso commented on src/pages/grants/GrantRecipesListPage.tsx:
```diff
@@ -1,17 +1,362 @@
+    } catch (error) {
+      console.error("Error fetching grant recipes:", error);
+      // Even if fetch fails, show the test recipe
+      setRecipes([testRecipe]);
```
Don't set if there is a system error.

jnakaso commented on src/pages/grants/GrantRecipesListPage.tsx:
```diff
@@ -1,17 +1,362 @@
+  };
+
+  const handleCellEditStop = async (params: GridCellEditStopParams) => {
+    if (!user) {
```
Remove.  Only authenticated users should reach this page

jnakaso commented on PR_DESCRIPTION.md:
1. Move to a document folder at the root
2. The title of the document it too broad  - tie to Jira ticket number

jnakaso reviewed (changes requested):
Please remove the upload portions of this PR.  There are no requirements to upload at this time.

jnakaso commented on src/pages/grants/GrantRecipesListPage.tsx:
```diff
@@ -1,17 +1,362 @@
+      return;
+    }
+
+    // Skip update for test recipe
```
This coding style is dangerous.  It is easy to overlook and could reach production.  Using test environment variables would help.

jnakaso commented on src/pages/grants/GrantRecipesListPage.tsx:
```diff
@@ -1,17 +1,362 @@
+      } else if (field === "modelType") {
+        updatedFields = { modelType: value as string };
+      } else if (field === "updatedAt") {
+        // Parse date string (MM/DD/YYYY) to Date
```
dayjs should be used.

jnakaso commented on PR_DESCRIPTION.md:
```diff
@@ -0,0 +1,118 @@
+   - Try editing all four fields on different rows
+   - Test date editing with MM/DD/YYYY format (e.g., "11/04/2025")
+
+3. **File Upload:**
```
Remove  file upload.

jnakaso commented on PR_DESCRIPTION.md:
```diff
@@ -0,0 +1,118 @@
+     - Date: 11/04/2025
+   - Verify all recipes from Firestore are displayed below the test recipe
+
+2. **Inline Editing:**
```
Please remove inline editing. 

1. validation is not enforced
2. certain fields are not even editable.
3. components can be used for some field

jnakaso commented on src/services/FirebaseStorageService.ts:
This is very good.  The application does not have requirements for file storage at this time, so let's remove it from the application.  The code can be implemented in the DAS component library.  Would you like to help there?

jnakaso commented on firestore.rules:
thanks for doing this

jnakaso committed:
Add notifications to system.

jnakaso committed:
Add notifications to system.

jnakaso committed:
Add notifications to system.

jnakaso committed:
lint

jnakaso committed:
lint

jnakaso committed:
Example code for auto-save

jnakaso committed:
lint fixes

jnakaso committed:
lint fixes

jnakaso committed:
lint fixes

jnakaso committed:
lint fixes

jnakaso committed:
addres comments

shardareddy12 committed:
feat: Implement grant recipe detail page with output fields (#7) #Wave62

- Add output fields section with field name and max count inputs
- Add toggle between word/char for output field limits
- Implement add/remove functionality for output fields
- Update GrantOutput type to support maxCount and unit (word/char)

shardareddy12 committed:
feat: Implement grant recipe detail page with output fields (#7) #Wave62

- Add output fields section with field name and max count inputs
- Add toggle between word/char for output field limits
- Implement add/remove functionality for output fields
- Update GrantOutput type to support maxCount and unit (word/char)

jnakaso committed:
address dev build inconsistencies

jnakaso committed:
saving output form

jnakaso committed:
remove log

ibarrasb committed:
proposal validation

jnakaso committed:
auto save every two seconds

jnakaso committed:
Grant recipe input editors

jnakaso committed:
Grant recipe input editors

jnakaso committed:
dev merge

jnakaso committed:
dev merge

jnakaso committed:
dev merge

jnakaso committed:
latest DAS libraries

jnakaso committed:
remove edits

jnakaso committed:
clean up proposals

jnakaso committed:
grid fixes

jnakaso committed:
Merge pull request #15 from digitalaidseattle/jeffn/grant-recipe-list

latest DAS libraries

jnakaso committed:
Merge branch 'dev' into feature/grant_recipe_list

jnakaso committed:
dev merge

jnakaso reviewed (approved):
let's more the description
